webapp/channels/src/components/interactive_dialog/interactive_dialog.test.tsx

```diff
@@ -121,11 +139,7 @@ describe('components/interactive_dialog/InteractiveDialog', () => {
         ];
 
         testCases.forEach((testCase) => test(`should interpret ${testCase.description}`, () => {
-            if (testCase.default === undefined) {
-                delete element.default;
-            } else {
-                element.default = testCase.default;
-            }
+            element.default = `${testCase.default}`;
```

Comment:
This is changing the test case as it was defined. It is not exactly the same logic, so I don't think it is testing exactly the same thing.

Reply:
basically first if check is not doing anything! because there is no `undefined` in the testcase
```ts
const testCases = [
            {description: 'no default', expectedChecked: false},
            {description: 'unknown default', default: 'unknown', expectedChecked: false},
            {description: 'default of "false"', default: 'false', expectedChecked: false},
            {description: 'default of true', default: true, expectedChecked: true},
            {description: 'default of "true"', default: 'True', expectedChecked: true},
            {description: 'default of "True"', default: 'True', expectedChecked: true},
            {description: 'default of "TRUE"', default: 'TRUE', expectedChecked: true},
        ];
``` 
![image](https://github.com/mattermost/mattermost/assets/93595710/1f62a854-156f-484a-99a2-67c6f5e1c024)
![image](https://github.com/mattermost/mattermost/assets/93595710/e9aa6b06-df24-4771-8bc9-12eccb69058f)

Reply:
Actually, the first test case has doesn't have the `default` value defined, so for the first test case `default` is `undefined`

Reply:
but the above mentioned error still persists! any solution to it ?

Reply:
@jespino ?